.tekton/.currency/scripts/generate_report.py

```diff
@@ -0,0 +1,202 @@
+
+    last_supported_version = re.search(
+        dependency + pattern, tekton_ci_output, flags=re.I | re.M
+    )
+
+    return last_supported_version[1]
+
+
+def isUptodate(last_supported_version, latest_version):
+    if last_supported_version == latest_version:
+        up_to_date = "Yes"
+    else:
+        up_to_date = "No"
+
+    return up_to_date
+
+
+def get_taskruns(namespace, task_name, taskrun_filter):
+    group = "tekton.dev"
+    version = "v1"
+    plural = "taskruns"
+
+    # access the custom resource from tekton
+    tektonV1 = client.CustomObjectsApi()
+    taskruns = tektonV1.list_namespaced_custom_object(
+        group,
+        version,
+        namespace,
+        plural,
+        label_selector=f"{group}/task={task_name}, triggers.tekton.dev/trigger=python-tracer-scheduled-pipeline-triggger",
+    )["items"]
+
+    filtered_taskruns = list(filter(taskrun_filter, taskruns))
+    filtered_taskruns.sort(
+        key=lambda tr: tr["metadata"]["creationTimestamp"], reverse=True
+    )
+
+    return filtered_taskruns
+
+
+def get_tekton_ci_output():
+    # config.load_kube_config()
+    config.load_incluster_config()
+
+    namespace = "default"
+
+    task_name = "python-tracer-unittest-gevent-starlette-task"
+    taskrun_filter = lambda tr: tr["status"]["conditions"][0]["type"] == "Succeeded"
+
+    starlette_taskruns = get_taskruns(namespace, task_name, taskrun_filter)
+
+    coreV1 = client.CoreV1Api()
+    tekton_ci_output = ""
+    for tr in starlette_taskruns:
+        pod_name = tr["status"]["podName"]
+        taskrun_name = tr["metadata"]["name"]
+        logs = coreV1.read_namespaced_pod_log(
+            pod_name, namespace, container="step-unittest"
+        )
+        if "Successfully installed" in logs:
+            print(
+                f"Retrieving container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}.."
+            )
+            match = re.search("Successfully installed .* (starlette-[^\s]+)", logs)
+            tekton_ci_output += f"{match[1]}\n"
+            break
+        else:
+            print(
+                f"Unable to retrieve container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}."
+            )
+
+    task_name = "python-tracer-unittest-default-task"
+    taskrun_filter = (
+        lambda tr: tr["metadata"]["name"].endswith("unittest-default-3")
+        and tr["status"]["conditions"][0]["type"] == "Succeeded"
+    )
+    default_taskruns = get_taskruns(namespace, task_name, taskrun_filter)
+
+    for tr in default_taskruns:
+        pod_name = tr["status"]["podName"]
+        taskrun_name = tr["metadata"]["name"]
+        logs = coreV1.read_namespaced_pod_log(
+            pod_name, namespace, container="step-unittest"
+        )
+        if "Successfully installed" in logs:
+            print(
+                f"Retrieving container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}.."
```

Comment:
These lines are essentially the same as 113-121, would it make sense to extract to a reusable function or lambda?

Reply:
Done